extensions/common/vault/vault-hashicorp/src/main/java/org/eclipse/edc/vault/hashicorp/HashicorpVaultClient.java

```diff
@@ -24,76 +25,206 @@
+        this.configValues = configValues;
+        this.headers = getHeaders();
+        this.healthCheckUrl = getHealthCheckUrl();
+    }
+
+    public Result<Void> doHealthCheck() {
+        var request = httpGet(healthCheckUrl);
+
+        try (var response = httpClient.execute(request)) {
+            if (response.isSuccessful()) {
+                return Result.success();
+            }
+
+            var code = response.code();
+            var errMsg = switch (code) {
+                case 429 -> "Vault is in standby";
+                case 472 -> "Vault is in recovery mode";
+                case 473 -> "Vault is in performance standby";
+                case 501 -> "Vault is not initialized";
+                case 503 -> "Vault is sealed";
+                default -> "Vault returned unspecified code %d".formatted(code);
+            };
+            var body = response.body();
+            if (body == null) {
+                return Result.failure("Healthcheck returned empty response body");
+            }
+            return Result.failure("Vault is not available. Reason: %s, additional information: %s".formatted(errMsg, body.string()));
+        } catch (IOException e) {
+            var errMsg = "Failed to perform healthcheck with reason: %s".formatted(e.getMessage());
+            monitor.warning(errMsg, e);
+            return Result.failure(errMsg);
+        }
+    }
+
+    /**
+     * Attempts to look up the Vault token defined in the configurations.
+     * <p>
+     * Will retry in some error cases.
+     *
+     * @return boolean indicating if the token is renewable
+     */
+    public Result<Boolean> lookUpToken() {
+        var uri = configValues.url()
+                .newBuilder()
+                .addPathSegment(TOKEN_LOOK_UP_SELF_PATH)
+                .build();
+        var request = httpGet(uri);
+
+        try (var response = httpClient.execute(request, FALLBACK_FACTORIES)) {
+            if (response.isSuccessful()) {
+                var responseBody = response.body();
+                if (responseBody == null) {
+                    return Result.failure("Token look up returned empty body");
+                }
+                var payload = objectMapper.readValue(responseBody.string(), new TypeReference<Map<String, Object>>() {});
+                var parseRenewableResult = parseRenewable(payload);
+                if (parseRenewableResult.failed()) {
+                    return Result.failure("Token look up response could not be parsed: %s".formatted(parseRenewableResult.getFailureDetail()));
+                }
+                var isRenewable = parseRenewableResult.getContent();
+                return Result.success(isRenewable);
+            } else {
+                return Result.failure("Token look up failed with status %d".formatted(response.code()));
+            }
+        } catch (IOException e) {
+            var errMsg = "Failed to look up token with reason: %s".formatted(e.getMessage());
```

Comment:
If thee error is reported back, there is no need to send it to the monitor here as that can be done by the client